crate/test_server/benches/benches.rs

```diff
@@ -0,0 +1,342 @@
+            hashing_algorithm: Some(HashingAlgorithm::SHA256),
+            ..Default::default()
+        },
+        "RSA AES KWP",
+    );
+}
+
+fn bench_rsa_encrypt(
+    c: &mut Criterion,
+    key_size: usize,
+    cryptographic_parameters: &CryptographicParameters,
+    name: &str,
+) {
+    let runtime = tokio::runtime::Runtime::new().unwrap();
+
+    let (kms_rest_client, _sk, pk) = runtime.block_on(async {
+        let ctx = ONCE
+            .get_or_try_init(start_default_test_kms_server)
+            .await
+            .unwrap();
+        let kms_rest_client = ctx.owner_client_conf.initialize_kms_client().unwrap();
+        let (sk, pk) = create_rsa_keypair(&kms_rest_client, key_size).await;
+        (kms_rest_client, sk, pk)
+    });
+
+    let mut group = c.benchmark_group("RSA tests");
+    group.bench_function(&format!("{name} {key_size}bit encryption"), |b| {
+        b.to_async(&runtime).iter(|| async {
+            let _ = encrypt(&kms_rest_client, &pk, &[0u8; 32], cryptographic_parameters).await;
+        });
+    });
+}
+
+fn bench_rsa_decrypt(
+    c: &mut Criterion,
+    key_size: usize,
+    cryptographic_parameters: &CryptographicParameters,
+    name: &str,
+) {
+    let runtime = tokio::runtime::Runtime::new().unwrap();
+
+    let (kms_rest_client, sk, _pk, ciphertext) = runtime.block_on(async {
+        let ctx = ONCE
+            .get_or_try_init(start_default_test_kms_server)
+            .await
+            .unwrap();
+        let kms_rest_client = ctx.owner_client_conf.initialize_kms_client().unwrap();
+        let (sk, pk) = create_rsa_keypair(&kms_rest_client, key_size).await;
+        let ciphertext = encrypt(&kms_rest_client, &pk, &[0u8; 32], cryptographic_parameters).await;
+        (kms_rest_client, sk, pk, ciphertext)
+    });
+
+    let mut group = c.benchmark_group("RSA tests");
+    group.bench_function(&format!("{name} {key_size}bit decryption"), |b| {
+        b.to_async(&runtime).iter(|| async {
+            let _ = decrypt(&kms_rest_client, &sk, &ciphertext, cryptographic_parameters).await;
+        });
+    });
+}
+
+async fn create_rsa_keypair(
+    kms_rest_client: &KmsClient,
+    cryptographic_length: usize,
+) -> (String, String) {
+    let create_key_pair_request =
+        create_rsa_key_pair_request(["bench"], cryptographic_length).unwrap();
+    // Query the KMS with your kmip data and get the key pair ids
+    let response = kms_rest_client
+        .create_key_pair(create_key_pair_request)
+        .await
+        .unwrap();
+    (
+        response.private_key_unique_identifier.to_string(),
+        response.public_key_unique_identifier.to_string(),
+    )
+}
+
+async fn encrypt(
+    kms_rest_client: &KmsClient,
+    pk: &str,
+    cleartext: &[u8],
+    cryptographic_parameters: &CryptographicParameters,
+) -> Vec<u8> {
+    // Create the kmip query
+    let encrypt_request = build_encryption_request(
+        pk,
+        None,
+        cleartext.to_vec(),
```

Comment:
Take ownership instead.

Reply:
done in f29feec55cba8edbe1e80ea9f53823f1d4603c27